### PR TITLE
Task: Spring Boot Portfolio Project — Part 2(DTO). Executor: Oleksii Bashkirov.

### PR DESCRIPTION
**Pull request by OleksiiBashkirov**
Improved project structure and functionality for online-book-store.
Refactored packages:
- config
- controller
- dto
- exception
- mapper
- model
- repository
- service

Edited: 
- BookService interface: updated methods save, findById and findAll.
- BookServiceImpl class: implemented methods for the updated interface.
- BookRepository interface: added Optional return type to the findById method.
- BookRepositoryImpl class: updated methods using jakarta.persistence library(EntityManagerFactory, EntityManager, EntityTransaction).
- Application class: removed CommandLineRunner bean.
- pom.xml file: added dependencies for mapstruct, mapstruct-processor,lombok-mapstruct-binding,mysql-connector-j and lombok. Configured maven-compiler-plugin.

Created:
- BookController class with @RestController, @RequestMapping(value = "/api/books"), and methods getAll, getBookById and createBook.
- BookDto class with fields: id, title, author, isbn, price, description and coverImage.
- CreateBookRequestDto class with the same fields.
- BookMapper interface with @Mapper(config = MapperConfig.class) annotation.
- EntityNotFoundException class.
- MapperConfig Class with @org.mapconstruct.MapperConfig annotation.

### CODE REVIEW

File: src/main/java/mate/academy/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

add `server.servlet.context-path=/api` to application.properties

File: src/main/java/mate/academy/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import java.math.BigDecimal;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.antlr.v4.runtime.misc.NotNull;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java]: Remove 
```
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
```

Add
```
@Getter
@Setter
```

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]:
```suggestion
                .toList();
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

**Review by okuzan — APPROVED** [review]

**Pull request merged**